src/Database/Oracle/Simple/ToField.hs

```diff
@@ -27,3 +31,28 @@ instance ToField Int where
 
 instance ToField DPITimestamp where
   toField = pure . AsTimestamp
+
+instance ToField UTCTime where
+  toField utcTime = pure $ AsTimestamp (utcTimeToDPITimestamp utcTime)
+
+utcTimeToDPITimestamp :: UTCTime -> DPITimestamp
+utcTimeToDPITimestamp utcTime = dpiTimeStampToUTCDPITimeStamp dpiTs
+  where
+    ZonedTime {..} = utcToZonedTime utc utcTime
+    LocalTime {..} = zonedTimeToLocalTime
+    (year, month, day) = toGregorian localDay
+    TimeOfDay {..} = localTimeOfDay
+    TimeZone {..} = zonedTimeZone
+    (seconds, fractionalSeconds) = properFraction todSec
+    (hourOffset, minuteOffset) = timeZoneMinutes `quotRem` 60
+    dpiTs = DPITimestamp
+      { year           = fromIntegral year
+      , month          = fromIntegral month
+      , day            = fromIntegral day
+      , hour           = fromIntegral todHour
+      , minute         = fromIntegral todMin
+      , second         = seconds
+      , fsecond        = truncate (fractionalSeconds * 1_000_000_000_000)
```

Comment:
`fsecond` in the struct expects nanos, not picos
https://odpi-c.readthedocs.io/en/latest/structs/dpiTimestamp.html

I'd convert from picos to nanos here.

Reply:
Dunzo